library/pod-security-policy/users/template.yaml

```diff
@@ -132,158 +132,341 @@ spec:
+            expression: |
+              variables.anyObject.kind == "Pod" ? 
+                (has(variables.anyObject.spec.securityContext) && has(variables.anyObject.spec.securityContext.runAsGroup) ? variables.anyObject.spec.securityContext.runAsGroup : null)
+              : null
+          - name: missingRequiredRunAsGroupContainers
+            expression: |
+              variables.badContainers.filter(container, 
+                has(variables.params.runAsGroup) && has(variables.params.runAsGroup.rule) && (variables.params.runAsGroup.rule == "MustRunAs") &&
+                !((has(container.securityContext) && has(container.securityContext.runAsGroup)) || 
+                  (variables.podRunAsGroup != null))
+              )
+          - name: invalidRunAsGroupContainers
+            expression: |
+              variables.badContainers.filter(container, 
+                !(variables.missingRequiredRunAsGroupContainers.exists(c, c.name == container.name)) &&
+                (
+                  has(variables.params.runAsGroup) && has(variables.params.runAsGroup.rule) ? 
+                  (
+                    variables.params.runAsGroup.rule == "RunAsAny" ? false :
+                    (
+                      (variables.params.runAsGroup.rule == "MustRunAs" || variables.params.runAsGroup.rule == "MayRunAs") && 
+                      (
+                        has(container.securityContext) && has(container.securityContext.runAsGroup) ? 
+                          !variables.params.runAsGroup.ranges.all(range, container.securityContext.runAsGroup >= range.min && container.securityContext.runAsGroup <= range.max) : 
+                          variables.podRunAsGroup == null || !variables.params.runAsGroup.ranges.all(range, variables.podRunAsGroup >= range.min && variables.podRunAsGroup <= range.max)
+                      )
+                    )
+                  )
+                  : false
+                )
+              )
+          - name: podRunAsFsGroup
+            expression: |
+              variables.anyObject.kind == "Pod" ? 
+                (has(variables.anyObject.spec.securityContext) && has(variables.anyObject.spec.securityContext.fsGroup) ? variables.anyObject.spec.securityContext.fsGroup : null)
+              : null
+          - name: missingRequiredFsGroupContainers
+            expression: |
+              variables.badContainers.filter(container, 
+                has(variables.params.fsGroup) && has(variables.params.fsGroup.rule) && (variables.params.fsGroup.rule == "MustRunAs") &&
+                !((has(container.securityContext) && has(container.securityContext.fsGroup)) || 
+                  (variables.podRunAsFsGroup != null))
+              )
+          - name: invalidRunAsFsGroupContainers
+            expression: |
+              variables.badContainers.filter(container, 
+                !(variables.missingRequiredFsGroupContainers.exists(c, c.name == container.name)) &&
```

Comment:
as above, let's not rely on names being globally unique

Reply:
Processing each container type separately.